system/jlib/jbuff.hpp

```diff
@@ -57,48 +57,72 @@ class jlib_decl MemoryAttr
 
 //--------------------------------------------------------------------------------------------------------------------
 
-template <class CLASS> class OwnedMalloc
+template <class CLASS, void (*FREE_FUNC)(CLASS *)> class OwnedPtrCustomFree
 {
+protected:
+    CLASS *ptr = nullptr;
+
 public:
-    inline OwnedMalloc()                        { ptr = NULL; }
-    inline OwnedMalloc(CLASS * _ptr)            { ptr = _ptr; }
-    explicit inline OwnedMalloc(unsigned n, bool clearMemory = false) { doAllocate(n, clearMemory); }
-    inline ~OwnedMalloc()                       { free(ptr); }
+    OwnedPtrCustomFree<CLASS, FREE_FUNC>() { }
+    OwnedPtrCustomFree<CLASS, FREE_FUNC>(CLASS *_ptr) : ptr(_ptr) { }
+    ~OwnedPtrCustomFree<CLASS, FREE_FUNC>() { FREE_FUNC(ptr); }
+
+    void operator = (CLASS * _ptr)
+    {
+        if (ptr)
+            FREE_FUNC(ptr);
+        ptr = _ptr;
+    }
+    inline CLASS * operator -> () const { return ptr; }
+    inline operator CLASS *() const     { return ptr; }
 
-    inline CLASS * operator -> () const         { return ptr; }
-    inline operator CLASS *() const             { return ptr; }
+    inline void clear()                 { CLASS *temp=ptr; ptr=nullptr; FREE_FUNC(temp); }
+    inline CLASS *get() const           { return ptr; }
+    inline CLASS *getClear()            { CLASS * temp = ptr; ptr=nullptr; return temp; }
+    inline void setown(CLASS *_ptr)     { if (ptr) FREE_FUNC(ptr); ptr = _ptr; }
 
-    inline void clear()                         { CLASS *temp=ptr; ptr=NULL; free(temp); }
-    inline CLASS * get() const                  { return ptr; }
-    inline CLASS * getClear()                   { CLASS * temp = ptr; ptr = NULL; return temp; }
-    inline void setown(CLASS * _ptr)            { CLASS * temp = ptr; ptr = _ptr; free(temp); }
+private:
+    inline OwnedPtrCustomFree(const OwnedPtrCustomFree<CLASS, FREE_FUNC> & other);
```

Comment:
better to say = delete;

Reply:
will change to public and delete to several methods